pkg/collector/corechecks/cluster/orchestrator/processors/chunking.go

```diff
@@ -0,0 +1,125 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build orchestrator
+// +build orchestrator
+
+package processors
+
+import (
+	"github.com/DataDog/datadog-agent/pkg/process/util"
+)
+
+// orchestratorMetadataList is a payload list of orchestratorMetadata payloads
+type orchestratorMetadataList struct {
+	orchestratorMetadataYaml [][]byte
+	orchestratorMetadata     []interface{}
+	chunker                  orchestratorMetadataChunker
+}
+
+// orchestratorMetadataChunker abstracts chunking of orchestratorMetadata payloads
+type orchestratorMetadataChunker interface {
+	// Accept takes a slice of orchestratorMetadata and allocates them to the current chunk
+	Accept(orchestratorMetadata []interface{}, weight int)
+}
+
+func (l *orchestratorMetadataList) Len() int {
+	return len(l.orchestratorMetadata)
+}
+
+func (l *orchestratorMetadataList) WeightAt(idx int) int {
```

Comment:
shouldn't the weight be additive of our resource representation + the yaml? I.e. if we have one representation that is almost a full copy of a large yaml we would have double the size?